src/subscriber.ts

```diff
@@ -212,6 +272,26 @@ export class Message {
       this._subscriber.nack(this);
     }
   }
+
+  /**
+   * Removes the message from our inventory and schedules it to be redelivered,
+   * with the modAck response being returned (for exactly once subscriptions).
+   *
+   * @example
+   * ```
+   * subscription.on('message', async (message) => {
+   *   const response = await message.nackWithResponse();
+   * });
+   * ```
+   */
+  async nackWithResponse(): Promise<AckResponse> {
+    if (!this._handled) {
```

Comment:
Same point as above: Can we short-circuit this in case exactly once delivery is disabled and return success?